File: api/app/Models/Parc.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;

class Parc extends Model
{
    use HasFactory;
    protected $fillable= ['nom','type','serie','fabricant','modele','version'];
/**
 * Get the technicien that owns the Parc
 *
 * @return \Illuminate\Database\Eloquent\Relations\BelongsTo
 */
public function technicien(): BelongsTo
{
    return $this->belongsTo(User::class);
}
}

File: lib/models/parc.dart
class Parc {
  final String nom;
  final int serie;
  final String type;
  final String fabricant;
  final String modele;
  final String version;

  const Parc(this.nom, this.type, this.serie, this.fabricant, this.modele, this.version);

  factory Parc.fromJson(Map<String, dynamic> json) {
    return Parc(
      json['nom'] as String,
      json['type'] as String,
      json['serie'] as int,
      json['fabricant'] as String,
      json['modele'] as String,
      json['version'] as String,
    
    );
  }
}
